server/server.js

```diff
@@ -2,25 +2,224 @@ import express from "express";
+app.get("/api/repos/:username/:repoName/commits", async (req, res) => {
+  const { username, repoName } = req.params;
+
+  try {
+    const url = `https://api.github.com/repos/${username}/${repoName}/commits?per_page=50`;
+
+    const response = await fetch(url, { headers: GH_HEADERS });
+
+    if (!response.ok) {
+      return res.status(response.status).json({ error: "GitHub API error" });
+    }
+
+    const data = await response.json();
+
+    const formatted = data.map((c) => ({
+      sha: c.sha,
+      message: c.commit.message,
+      date: c.commit.author?.date,
+      author: c.commit.author?.name,
+    }));
+
+    res.json(formatted);
+  } catch (error) {
+    console.error("Commits error:", error);
+    res.status(500).json({ error: "Failed to fetch commits" });
+  }
+});
+
+/* -----------------------------------------------------------
+   GET README.md as raw markdown
+   /api/repos/:username/:repoName/readme
+----------------------------------------------------------- */
+app.get("/api/repos/:username/:repoName/readme", async (req, res) => {
+  const { username, repoName } = req.params;
+
+  try {
+    const url = `https://api.github.com/repos/${username}/${repoName}/readme`;
+
+    const response = await fetch(url, { headers: GH_HEADERS });
+
+    if (response.status === 404) {
+      return res.json({ content: "# No README.md found" });
+    }
+
+    if (!response.ok) {
+      return res.status(response.status).json({ error: "GitHub API error" });
+    }
+
+    const data = await response.json();
+
+    const markdown = Buffer.from(data.content, "base64").toString("utf-8");
+
+    res.json({ content: markdown });
+  } catch (error) {
+    console.error("README error:", error);
+    res.status(500).json({ error: "Failed to fetch README" });
+  }
+});
+
+/* -----------------------------------------------------------
+   SEARCH repos: /api/search?query=...
+----------------------------------------------------------- */
+app.get("/api/search", async (req, res) => {
+  const query = (req.query.query || "").toLowerCase();
+
+  try {
+    const response = await fetch(
+      "https://api.github.com/user/repos?per_page=200",
+      { headers: GH_HEADERS }
+    );
+
+    const repos = await response.json();
+
+    const filtered = repos.filter((r) =>
+      r.name.toLowerCase().includes(query)
+    );
+
+    res.json(filtered);
+  } catch (error) {
+    console.error("Search error:", error);
+    res.status(500).json({ error: "Failed to search repos" });
+  }
+});
+
+/**
+ * POST /api/generate-blog
+ * Body: { username: string, repo: string, commits: Array<{sha, message, author, date}> (optional) }
+ */
+
+
+app.post("/api/generate-blog", async (req, res) => {
+  const { username, repo, commits } = req.body;
+
+  if (!username || !repo) {
+    return res.status(400).json({ error: "username and repo are required" });
+  }
+
+  try {
+    // Fetch recent commits if not passed in
+    let commitList = commits;
+    if (!commitList) {
```

Comment:
```suggestion
    if (!commits) {
```

하고 아래는 const keyword를 사용할 수 있지 않을까요?